src/robot/libdocpkg/xmlwriter.py

```diff
@@ -15,33 +15,51 @@
 
 from robot.utils import XmlWriter, get_timestamp
 
+from .htmlwriter import DocToHtml
+
 
 class LibdocXmlWriter(object):
 
+    def __init__(self, html_doc_format=False):
+        self._html_doc_format = html_doc_format
+        self._formatter = None
+        self._writer = None
+
     def write(self, libdoc, outfile):
-        writer = XmlWriter(outfile)
-        writer.start('keywordspec', {'name': libdoc.name, 'type': libdoc.type,
-                                     'format': libdoc.doc_format,
-                                     'generated': get_timestamp(millissep=None)})
-        writer.element('version', libdoc.version)
-        writer.element('scope', libdoc.scope)
-        writer.element('namedargs', 'yes' if libdoc.named_args else 'no')
-        writer.element('doc', libdoc.doc)
-        self._write_keywords('init', libdoc.inits, writer)
-        self._write_keywords('kw', libdoc.keywords, writer)
-        writer.end('keywordspec')
-        writer.close()
-
-    def _write_keywords(self, type, keywords, writer):
+        self._formatter = DocToHtml(libdoc.doc_format) if self._html_doc_format else self.raw_format
+        self._writer = XmlWriter(outfile)
```

Comment:
I don't like setting values to `self` when they could just be passed to functions as arguments. The latter approach worked fine earlier with `writer` and I don't see why it wouldn't work with `formatter` as well. Using `self` for this purpose feels like misusing object state and `writer.start(...)` also reads better than `self._writer.start(...)`.

Also, we try to keep the code PEP-8 compliant and that includes keeping lines less than 80 characters long. The line with `formatter` should either be split or a separate `DocFormatter` implemented and logic moved there.

Reply:
done